### ISSUE TITLE
Rejects `let` as for-statement body

### ISSUE THREAD

**Issue by bakkot** [shapesecurity/shift-parser-js#389]
See [this test](https://github.com/tc39/test262/blob/master/test/language/statements/for/let-block-with-newline.js).

```js
for (; false; ) let
{}
```

is a perfectly valid program: it is a `for` statement whose body is an ExpressionStatement consisting of an `IdentifierExpression`, followed by an empty block.

Because there is no lookahead restriction for `let {` for ExpressionStatements it can start parsing as one, and because there is no valid parse beginning `for (; false; ) let \n {` (unlike at the top level of a script), [ASI](https://tc39.github.io/ecma262/#sec-automatic-semicolon-insertion) kicks in after the `let`.

**Comment by bakkot** [comment]
Ditto for

```js
for (; false; ) let
x = 0;
```
```js
for (var x in null) let
{}
```
```js
for (var x in null) let
x = 0;
```
```js
for (var x of []) let
{}
```
```js
for (var x of []) let
x = 0
```
```js
if (false) let
{}
```
```js
if (false) let
x = 0
```
```js
l: let
{}
```
```js
l: let
x = 0
```
```js
with ({}) let
{}
```
```js
with ({}) let
x = 0
```
```js
while (false) let
{}
```
```js
while (false) let
x = 0
```

**Comment by Protryon** [comment]
Addressed by #396.
